test/sol5/kyberNetwork.js

```diff
@@ -1091,6 +1092,148 @@ contract('KyberNetwork', function(accounts) {
+                "!network"
+            );
+            await network.setEnable(true, { from: admin });
+        });
+    
+        it("test can not trade when caller is not proxy", async () => {
+            let notAProxy = accounts[9];
+            let proxies = await network.getKyberProxies();
+            Helper.assertEqual(proxies.length, 1);
+            assert.notEqual(proxies, notAProxy)
+    
+            // calling network from non proxy, expect revert
+            await expectRevert(
+                network.tradeWithHintAndFee(networkProxy, srcToken.address, srcQty, ethAddress, taker,
+                    maxDestAmt, minConversionRate, platformWallet, platformFee, emptyHint, { from: notAProxy }),
+                "bad sender"
+            );
+        });
+    
+        it.skip("test can not trade when setting gas price too high", async () => {
+            let invalidGasPrice = gasPrice.add(new BN(1));
+    
+            await expectRevert(
+                network.tradeWithHintAndFee(networkProxy, srcToken.address, srcQty, ethAddress, taker,
+                    maxDestAmt, minConversionRate, platformWallet, platformFee, emptyHint, { gasPrice: invalidGasPrice }),
+                "gas price"
+            );
+        });
+    
+        it("test can not trade when src amount too high", async () => {
+            let invalidSrcQty = KYBER_MAX_QTY.add(new BN(1));
+    
+            await expectRevert(
+                network.tradeWithHintAndFee(networkProxy, srcToken.address, invalidSrcQty, ethAddress, taker,
+                    maxDestAmt, minConversionRate, platformWallet, platformFee, emptyHint),
+                "srcAmt > MAX_QTY"
+            );
+        });
+    
+        it("test can not trade when src amount is zero", async () => {
+            let invalidSrcQty = new BN(0);
+    
+            await expectRevert(
+                network.tradeWithHintAndFee(networkProxy, srcToken.address, invalidSrcQty, ethAddress, taker,
+                    maxDestAmt, minConversionRate, platformWallet, platformFee, emptyHint),
+                "0 srcAmt"
+            );
+        });
+    
+        it("test can not trade when dst addr is 0", async () => {
+            let invalidTaker = zeroAddress;
+    
+            await expectRevert(
+                network.tradeWithHintAndFee(networkProxy, srcToken.address, srcQty, ethAddress, invalidTaker,
+                    maxDestAmt, minConversionRate, platformWallet, platformFee, emptyHint),
+                "dest add 0"
+            );
+        });
+    
+        it("test can not trade when src === dst", async () => {
+            await expectRevert(
+                network.tradeWithHintAndFee(networkProxy, srcToken.address, srcQty, srcToken.address, taker,
+                    maxDestAmt, minConversionRate, platformWallet, platformFee, emptyHint),
+                "src = dest"
+            );
+        });
+    
+        it("test can not trade when fee is to high", async () => {
```

Comment:
typo
to --> too

Reply:
resolved in 8a85cb2840731bfd6de26d589980f04b587331c4